src/coreclr/vm/syncblk.cpp

```diff
@@ -1567,6 +1571,19 @@ AwareLock::EnterHelperResult ObjHeader::EnterObjMonitorHelperSpin(Thread* pCurTh
         return AwareLock::EnterHelperResult_Contention;
     }
 
+    // Initialize g_currentProcessCpuCount and g_currentProcessIsCpuQuotaLimited if they haven't been initialized yet
+    if (g_currentProcessCpuCount == 0)
+    {
+        GetCurrentProcessCpuCount();
+    }
+
+    _ASSERTE(g_currentProcessCpuCount > 0);
+    if (g_currentProcessIsCpuQuotaLimited)
+    {
+        g_SpinConstants.dwMonitorSpinCount = 0;
+        return AwareLock::EnterHelperResult_Contention;
+    }
```

Comment:
This should be based on the config option to spin-wait or not in CPU quota-limited environments. I think CoreCLR can set up the config var as it typically does and the managed side can check for it separately using only the `DOTNET_` prefix, since it's a boolean value and a new config var.